src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -0,0 +1,907 @@
+                    }
+
+                    var stringVal = jTokenExpression.Value.Value<string>()!;
+                    var resolved = nameResolver.TryLookupName(NameType.Variable, stringVal) ?? throw new ArgumentException($"Unable to find variable {stringVal}");
+                    baseSyntax = SyntaxHelpers.CreateIdentifier(resolved);
+                    break;
+                }
+                case "reference":
+                {
+                    if (expression.Parameters.Length == 1 && expression.Parameters[0] is FunctionExpression resourceIdExpression && resourceIdExpression.NameEquals("resourceid"))
+                    {
+                        // resourceid directly inside a reference - check if it's a reference to a known resource
+                        var resourceName = TryLookupResource(resourceIdExpression);
+                        
+                        if (resourceName != null)
+                        {
+                            baseSyntax = new PropertyAccessSyntax(
+                                new VariableAccessSyntax(SyntaxHelpers.CreateIdentifier(resourceName)),
+                                SyntaxHelpers.CreateToken(TokenType.Dot, "."),
+                                SyntaxHelpers.CreateIdentifier("properties"));
+                        }
+                    }
+                    break;
+                }
+                case "resourceid":
+                {
+                    var resourceName = TryLookupResource(expression);
+
+                    if (resourceName != null)
+                    {
+                        baseSyntax = new PropertyAccessSyntax(
+                            SyntaxHelpers.CreateIdentifier(resourceName),
+                            SyntaxHelpers.CreateToken(TokenType.Dot, "."),
+                            SyntaxHelpers.CreateIdentifier("id"));
+                    }
+                    break;
+                }
+                case "format":
+                {
+                    baseSyntax = TryParseStringExpression(expression);
+                    break;
+                }
+                case "concat":
+                {
+                    var canInterpolate = expression.Parameters.Any(p => p is JTokenExpression jTokenExpression && jTokenExpression.Value.Type == JTokenType.String);
+                    if (!canInterpolate)
+                    {
+                        // we might be dealing with an array
+                        break;
+                    }
+
+                    if (expression.Parameters.Length < 1)
+                    {
+                        // empty concat statement - let's just ignore that...
+                        break;
+                    }
+
+                    baseSyntax = TryParseStringExpression(expression);
+                    break;
+                }
+                case "__bicep_reference":
```

Comment:
What's this?

Reply:
I killed it. I was using it to store info to pass to a later part of the decompilation, but it's no longer necessary.